Firestore/core/test/unit/local/leveldb_index_manager_test.cc

```diff
@@ -62,11 +71,1042 @@ class LevelDbIndexManagerTest : public ::testing::Test {
+
+    auto q = Query("coll");
+
+    std::vector<std::pair<core::Query, std::vector<std::string>>> test_cases = {
+        {q.AddingOrderBy(OrderBy("int")),
+         {"coll/{array:[1,foo],int:1}", "coll/{array:[3,foo],int:3}"}},
+        {q.AddingFilter(
+             Filter("float", "==", std::numeric_limits<double>::quiet_NaN())),
+         {"coll/{float:nan}"}},
+        {q.AddingFilter(Filter("float", "==", -0.0)),
+         {"coll/{float:-0.0,string:a}", "coll/{float:0,string:ab}",
+          "coll/{float:0.0,string:b}"}},
+        {q.AddingFilter(Filter("float", "==", 0)),
+         {"coll/{float:-0.0,string:a}", "coll/{float:0,string:ab}",
+          "coll/{float:0.0,string:b}"}},
+        {q.AddingFilter(Filter("float", "==", 0.0)),
+         {"coll/{float:-0.0,string:a}", "coll/{float:0,string:ab}",
+          "coll/{float:0.0,string:b}"}},
+        {q.AddingFilter(Filter("string", "==", "a")),
+         {"coll/{float:-0.0,string:a}"}},
+        {q.AddingFilter(Filter("string", ">", "a")),
+         {"coll/{float:0,string:ab}", "coll/{float:0.0,string:b}"}},
+        {q.AddingFilter(Filter("string", ">=", "a")),
+         {"coll/{float:-0.0,string:a}", "coll/{float:0,string:ab}",
+          "coll/{float:0.0,string:b}"}},
+        {q.AddingFilter(Filter("string", "<", "b")),
+         {"coll/{float:-0.0,string:a}", "coll/{float:0,string:ab}"}},
+        {q.AddingFilter(Filter("string", "<", "coll")),
+         {"coll/{float:-0.0,string:a}", "coll/{float:0,string:ab}",
+          "coll/{float:0.0,string:b}"}},
+        {q.AddingFilter(Filter("string", ">", "a"))
+             .AddingFilter(Filter("string", "<", "b")),
+         {"coll/{float:0,string:ab}"}},
+        {q.AddingFilter(Filter("array", "array-contains", "foo")),
+         {"coll/{array:[1,foo],int:1}", "coll/{array:[2,foo]}",
+          "coll/{array:[3,foo],int:3}"}},
+        {q.AddingFilter(Filter("array", "array-contains-any", Array(1, "foo"))),
+         {"coll/{array:[1,foo],int:1}", "coll/{array:[1]}",
+          "coll/{array:[2,foo]}", "coll/{array:[3,foo],int:3}"}},
+        {q.AddingFilter(Filter("multi", ">=", true)), {"coll/{multi:true}"}},
+        {q.AddingFilter(Filter("multi", ">=", 0)), {"coll/{multi:1}"}},
+        {q.AddingFilter(Filter("multi", ">=", "")), {"coll/{multi:string}"}},
+        {q.AddingFilter(Filter("multi", ">=", Array())), {"coll/{multi:[]}"}},
+        {q.AddingFilter(Filter("multi", "!=", true)),
+         {"coll/{multi:1}", "coll/{multi:string}", "coll/{multi:[]}"}},
+        {q.AddingFilter(Filter("multi", "in", Array(true, 1))),
+         {"coll/{multi:true}", "coll/{multi:1}"}},
+        {q.AddingFilter(Filter("multi", "not-in", Array(true, 1))),
+         {"coll/{multi:string}", "coll/{multi:[]}"}},
+        {q.AddingOrderBy(OrderBy("array"))
+             .StartingAt(Bound::FromValue(Array(Array(2)), true)),
+         {"coll/{array:[2,foo]}", "coll/{array:[3,foo],int:3}"}},
+
+        {q.AddingOrderBy(OrderBy("array", "desc"))
```

Comment:
I am not quite able to decipher the pattern you used for adding newlines, but if there is one, feel free to keep as is:)